### PR TITLE
[ezfoundation] Add new port

### PR DESCRIPTION
**Pull request by tkeinz**
**Describe the pull request**

- #### What does your PR fix?
  Adds a new port: ezfoundation

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?
  windows;uwp, Yes

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
  Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
  Yes

### CODE REVIEW

File: ports/ezfoundation/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  REPO ezEngine/ezEngine
+  REF 1448308ab99bae0d8d2a292ad345ff3a757f3124 #21.10
+  SHA512 20da87772366d1612795f534b31600123c0d04beba252d27d895c3c3cff7dae65952bd890629d79a7a691c2f3444601552c4b4eca99e8f8a99dc935ce2ebd284
+  PATCHES
+    ezFoundation_vcpkg.patch
+)
+
+vcpkg_cmake_configure(
+  SOURCE_PATH "${SOURCE_PATH}"
+  GENERATOR Ninja
```
Comment (JonLiu1993) [inline ports/ezfoundation/portfile.cmake:14]:
```suggestion
```

File: scripts/ci.baseline.txt [hunk 2]
```diff
@@ -227,6 +227,8 @@ embree2:x64-linux=skip
 embree2:x64-osx=skip
 epsilon:arm-uwp=fail
 epsilon:x64-uwp=fail
+ezfoundation:x64-linux=skip
+ezfoundation:x64-osx=skip
```
Comment (JonLiu1993) [inline scripts/ci.baseline.txt:231]: @tkeinz, Thanks for your pr, Can you confirm that ezfoundation does not officially support these two platforms? If it is not supported, please use the "supports" field in vcpkg.json to indicate

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]

File: scripts/ci.baseline.txt (hunk 2 above)
Comment (markkeinz) [reply scripts/ci.baseline.txt]: Thanks for your review!

Yes, at this time, only Windows and UWP are officially supported. As you suggested, I added a "supports" field to vcpkg.json and removed the CI Baseline entries again.

Comment (tkeinz) [reply scripts/ci.baseline.txt]: See above comment - sorry, was logged in with my work account and didn't notice when posting that comment.

File: ports/ezfoundation/portfile.cmake (hunk 1 above)
Comment (tkeinz) [reply ports/ezfoundation/portfile.cmake]: Removed as suggested.

**Review by JonLiu1993 — APPROVED** [review]

**Pull request merged**